File: P1/FactoriaAbstracta/src/factoriaabstracta/FactoriaMontana.java
package factoriaabstracta;

public class FactoriaMontana extends FactoriaCarreraYBicicleta {

        @Override
	public CarreraMontana crearCarrera(int i) 
        {
		return new CarreraMontana(i);
	}

        @Override
	public BicicletaMontana crearBicicleta(int i) 
        {
		return new BicicletaMontana(i);
	}
}

File: P1/S1E2/lib/factoria_montana.rb
require './carrera_montana.rb'
require './bicicleta_montana.rb'

module PrototipoRuby
  class FactoriaMontana
    
    def crearCarrera(id)
      return CarreraMontana.new(id)
    end
    
    def crearBicicleta(id)
      return BicicletaMontana.new(id)
    end
  end
end
